packages/strapi-plugin-content-type-builder/admin/src/containers/DataManagerProvider/reducer.js

```diff
@@ -24,6 +25,47 @@ const getOppositeNature = originalNature => {
   }
 };
 
+const addComponentsToState = (state, componentToAddUid, objToUpdate) => {
+  let newObj = objToUpdate;
+  const componentToAdd = state.getIn(['components', componentToAddUid]);
+  const isTemporaryComponent = componentToAdd.get('isTemporary');
+  const componentToAddSchema = componentToAdd.getIn(['schema', 'attributes']);
+  const hasComponentAlreadyBeenAdded =
+    state.getIn(['modifiedData', 'components', componentToAddUid]) !==
+    undefined;
+
+  // created components are already in the modifiedData.components
+  // We don't add them because all modifications will be lost
+  if (isTemporaryComponent || hasComponentAlreadyBeenAdded) {
+    return newObj;
+  }
+
+  // Add the added components to the modifiedData.compontnes
+  newObj = newObj.set(componentToAddUid, componentToAdd);
+  const nestedComponents = retrieveComponentsFromSchema(
+    componentToAddSchema.toJS(),
+    state.get('components').toJS()
+  );
+
+  // We need to add the nested components to the modifiedData.components as well
+  nestedComponents.forEach(componentUid => {
+    const isTemporary =
+      state.getIn(['components', componentUid, 'isTemporary']) || false;
```

Comment:
`state.getIn(['components', componentUid, 'isTemporary'], false)` can be used

Reply:
It's immutable and not lodash so you can't pass a default value...

Reply:
Okay, my bad!